File: library/src/main/java/zhan/library/delegate/ScrollerDelegate.java
package zhan.library.delegate;

import android.view.View;
import android.widget.Scroller;


public class ScrollerDelegate {

  private Scroller mScroller;
  private View mTargetView;

  public ScrollerDelegate(View targetView) {
    mTargetView = targetView;
    mScroller = new Scroller(targetView.getContext());
  }

  public void computeScroll() {
    if (mScroller.computeScrollOffset()) {
      mTargetView.scrollTo(mScroller.getCurrX(), mScroller.getCurrY());
      mTargetView.postInvalidate();
    }
  }

  public void smoothScrollTo(int startX, int startY,
      int dx, int dy, int duration) {
    mScroller.startScroll(startX, startY, dx, dy, duration);
    mTargetView.invalidate();
  }
}

File: double-pull-delegate/src/main/java/zhan/pull/delegate/ScrollerDelegate.kt
package zhan.pull.delegate

import android.view.View
import android.widget.Scroller


open class ScrollerDelegate(private val mTargetView: View) {

    private val mScroller: Scroller

    init {
        mScroller = Scroller(mTargetView.context)
    }

    fun computeScroll() {
        if (mScroller.computeScrollOffset()) {
            mTargetView.scrollTo(mScroller.currX, mScroller.currY)
            mTargetView.postInvalidate()
        }
    }

    fun smoothScrollTo(startX: Int, startY: Int,
                       dx: Int, dy: Int, duration: Int) {
        mScroller.startScroll(startX, startY, dx, dy, duration)
        mTargetView.invalidate()
    }
}
